contrib/migrate/migrate.c

```diff
@@ -0,0 +1,897 @@
+     case \
+       when class_type = 0 then \
+      'rename table [' || class_name || '] to [' || lower (owner.name) || '.' || class_name || '] ' \
+     else \
+      'rename view [' || class_name || '] to [' || lower (owner.name) || '.' || class_name || '] ' \
+     end as q \
+   from \
+     _db_class \
+   where \
+     is_system_class % 8 = 0";
+
+/* update class_name and unique_name except for system classes. */
+static char *update_db_class_not_for_system_classes =
+  "update _db_class set class_name = substring_index (class_name, '.', -1), unique_name = class_name where is_system_class % 8 = 0";
+
+static char *serial_query = {
+  "select \
+      'call change_serial_owner (''' || name || ''', ''' || substring_index (name, '.', 1) || ''') on class db_serial' as q \
+   from db_serial \
+   where class_name is not null and instr (name, '.') > 0"
+};
+
+static char *update_serial[] = {
+  "update db_serial set name = substring_index (name, '.', -1)",
+  "update db_serial set unique_name = lower (owner.name) || '.' || name"
+};
+
+static char *update_trigger = "update db_trigger set unique_name = lower (owner.name) || '.' || name";
+
+static char *index_query[] = {
+  "create unique index u_db_serial_name_owner ON db_serial (name, owner)",
+  "alter table db_serial drop constraint pk_db_serial_name",
+  "alter table db_serial add constraint pk_db_serial_unique_name primary key (unique_name)",
+
+  "create index i__db_class_unique_name on _db_class (unique_name)",
+  "create index i__db_class_class_name_owner on _db_class (class_name, owner)",
+
+  "drop index i__db_class_class_name on _db_class"
+};
+
+/* only system classes update unique_name. */
+static char *update_db_class_for_system_classes =
+  "update _db_class set unique_name = class_name where is_system_class % 8 != 0";
+
+static void
+print_errmsg (const char *err_msg)
+{
+  fprintf (stderr, "ERROR: %s\n", err_msg);
+
+  return;
+}
+
+static void
+print_log (const char *log_fmt, ...)
+{
+  char log_msg[BUF_LEN];
+  time_t t;
+  struct tm *tm;
+  va_list ap;
+
+  t = time (NULL);
+  tm = localtime (&t);
+
+  snprintf (log_msg, BUF_LEN, "%d-%02d-%02d %02d:%02d:%02d\t",
+	    tm->tm_year + 1900, tm->tm_mon + 1, tm->tm_mday, tm->tm_hour, tm->tm_min, tm->tm_sec);
+
+  va_start (ap, log_fmt);
+
+  vsnprintf (log_msg + strlen (log_msg), BUF_LEN - strlen (log_msg), log_fmt, ap);
+
+  va_end (ap);
+
+  snprintf (log_msg + strlen (log_msg), BUF_LEN - strlen (log_msg), "\n");
+
+  fprintf (stderr, "ERROR : %s\n", log_msg);
+
+  return;
+}
+
+static int
+char_isspace (int c)
+{
+  return ((c) == ' ' || (c) == '\t' || (c) == '\r' || (c) == '\n' || (c) == '\f' || (c) == '\v');
+}
+
+static char *
+next_char (char *str_p)
+{
+  char *p;
+
+  p = str_p;
+  while (char_isspace ((int) *p) && *p != '\0')
+    {
+      p++;
+    }
+
+  return (p);
+}
+
+static char *
+get_token (char *str_p, char **token_p)
+{
+  char *p, *end, *token = NULL;
+  int length;
+
+  token = NULL;
```

Comment:
It's trivial
We only need to initialize it once.

Reply:
Fixed, thanks